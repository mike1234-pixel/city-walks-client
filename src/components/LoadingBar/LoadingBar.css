/* loading bar */
.loading-bar {
  width: 100%;
  height: 40px;
  background: var(--linear-blue);
  animation: loading 2s linear;
  animation-iteration-count: infinite;
  border-radius: 5px;
}
.loading-bar__container {
  height: 90vh;
  margin-top: 30px;
}

@keyframes loading {
  0% {
    width: 0%;
  }
  100% {
    width: 100%;
  }
}

/*# sourceMappingURL=LoadingBar.css.map */
