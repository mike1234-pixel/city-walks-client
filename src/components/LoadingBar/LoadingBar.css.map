{"version":3,"sourceRoot":"","sources":["LoadingBar.scss"],"names":[],"mappings":"AAAA;AAEA;EACI;EACA;EACA;EACA;EACA;EACA;;AAEA;EACI;EACA;;;AAIR;EACI;IACI;;EAGJ;IACI","file":"LoadingBar.css"}