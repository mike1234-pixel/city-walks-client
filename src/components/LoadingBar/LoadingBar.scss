/* loading bar */

.loading-bar {
    margin-top: 8px;
    width: 100%;
    height: 40px;
    background: linear-gradient(159deg, rgb(0, 96, 134) 0%, rgb(64, 156, 211) 100%);
    animation: loading 2s linear;
    animation-iteration-count: infinite;

    &__container {
        height: 90vh;
    }
}

@keyframes loading {
    0% {
        width: 0%;
    }

    100% {
        width: 100%;
    }
}