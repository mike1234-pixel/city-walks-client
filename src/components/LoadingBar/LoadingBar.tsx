import React from 'react'
import './LoadingBar.scss'

const LoadingBar: React.FC = () => {
    return (
        <div className="loading-bar__container">
            <p>Loading...</p>
            <div className="loading-bar"></div>
        </div>
    )
}

export default LoadingBar

