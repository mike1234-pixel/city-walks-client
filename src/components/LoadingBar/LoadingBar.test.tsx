import renderer, { ReactTestRendererJSON } from 'react-test-renderer'
import { cleanup, render } from '@testing-library/react'
import LoadingBar from './LoadingBar'

describe('Test Loading Bar', () => {

    afterEach(cleanup);

    // snapshot test
    it('Loading Bar renders correctly', () => {

        const tree: ReactTestRendererJSON | ReactTestRendererJSON[] | null = renderer
            .create(<LoadingBar />)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });

    // unit tests
    it('Loading Bar has text content', () => {
        const { getByText } = render(<LoadingBar />)

        const slideText: HTMLElement = getByText("Loading...")

        expect(slideText).toBeDefined()

    })
})