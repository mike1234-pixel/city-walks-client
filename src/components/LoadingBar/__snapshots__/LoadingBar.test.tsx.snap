// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Loading Bar Loading Bar renders correctly 1`] = `
<div
  className="loading-bar__container"
>
  <p>
    Loading...
  </p>
  <div
    className="loading-bar"
  />
</div>
`;
